### ISSUE TITLE
Trade - max button on RUNE trade pulls incorrect USD value

### ISSUE THREAD

**Issue by SS-FOX-McCloud** [shapeshift/web#568]
When setting up a trade from RUNE (ERC20) to any other asset the fiat value of the trade is incorrectly displayed at the top of the trade modal. 

- visit app.shapeshift.com
- in the trade modal select RUNE (ERC20) as the input coin
- select any coin as the output
- click "max" for the RUNE amount
- Notice the total value of RUNE is incorrect in fiat terms 

<img width="367" alt="Screen_Shot_2021-12-14_at_11 34 43_AM" src="https://user-images.githubusercontent.com/57505863/146060773-c6a61ca5-2aed-4570-a30e-b8c38551aa64.png">

A/C:
RUNE (ERC20) value is displayed correctly, in fiat,  when it is the input of a trade.

**Comment by chxash** [comment]
I investigated this one and found that some swapper.getUsdRate() call returns incorrect values for RUNE / USDC (it's also present for other USDC pairs).
The call can be found here in web : https://github.com/shapeshift/web/blob/develop/src/components/Trade/hooks/useSwapper/useSwapper.ts#L271

In the lib, getUsdRate is performing a call to 0x api, we can reproduce the api calls with following links : 

For 1USDC like in the lib :
https://api.0x.org/swap/v1/quote?buyToken=USDC&buyAmount=1000000&sellToken=0x3155ba85d5f96b2d030a4966af206230e46849cb

This pricing is incorrect relative to current RUNE price ...

For 100USDC (can be also tried with more) : 

https://api.0x.org/swap/v1/quote?buyToken=USDC&buyAmount=100000000&sellToken=0x3155ba85d5f96b2d030a4966af206230e46849cb

The pricing here is more correct per RUNE.

Basically in the lib we are fetching price for 1USDC and it leads to incorrect values for some assets, do we want to raise an issue in the lib regarding this ? or maybe investigate more on the 0x api side why price is so weird for 1 USDC.

**Comment by 0xean** [comment]
👏  - thanks @chxash for the work here to dive deep and get this resolved.  Feel free to send me a message with hours spent and I can tip you out for it.  

Again, very much appreciate all of your recent PRs!  Keep em comin.

**Comment by chxash** [comment]
Integrated lib fix within https://github.com/shapeshift/web/pull/1036

Bug is fixed in web with this : 

![rune_fixed](https://user-images.githubusercontent.com/5720927/154449406-c34a8b75-8660-459c-88fc-5020a61284ab.png)

**Comment by chxash** [comment]
Bug fix from lib has been integrated into web develop branch, this issue can be verified / closed.

**Comment by 0xean** [comment]
@SS-FOX-McCloud - can you please verify against develop and we can close this once done?
